ZCS-15980: updated timezone data with Sep 2024 data

k-kato opened the pull request:
Update timezone data with `tzdata2024b` provided by iana.

Notes:
* `%z` has been introduced for time zone abbreviations.
   * See https://github.com/Zimbra/zm-mailbox/pull/1668
* new timezone `America/Mazatlan` and `America/Nuuk` are added
* `America/Chihuahua` is now included in `Central Standard Time (Mexico)` Windows time zone id.
   * See https://techcommunity.microsoft.com/t5/daylight-saving-time-time-zone/mexico-2023-time-zone-updates-now-available/bc-p/3779557/highlight/true
* `UTC+03:00 Asia/Damascus` has been applied to TzMsg files in https://github.com/Zimbra/zm-timezones/pull/31

Additional changes:
* added `fixICS.pl` to fix `Europe/Dublin` timezone in `timezones.ics`
* updated `timezones.ics` using the script. Now there is no difference in `TZID:Europe/Dublin` section. (See `timezones.ics`, new lines 3727-3746)
* updated readme

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Zimbra/zm-timezones?pullRequest=37) <br/>All committers have signed the CLA.

k-kato commented on conf/tz/tz-src-url.txt:
```diff
@@ -1 +1 @@
-ftp://ftp.iana.org/tz/releases/tzdata2024a.tar.gz
+https://data.iana.org/time-zones/releases/tzdata2024b.tar.gz
```
Changed to `https` as ftp protocol is no longer supported on major browsers.

silentsakky reviewed (approved):
Changes looks good, just cross check https://github.com/Zimbra/zm-timezones/commit/e8d9a278ff5daad791d2109d621be3ce2b7f2ec4 is preserved as we added it as a manual change

k-kato commented:
> Changes looks good, just cross check [e8d9a27](https://github.com/Zimbra/zm-timezones/commit/e8d9a278ff5daad791d2109d621be3ce2b7f2ec4) is preserved as we added it as a manual change

checking...

k-kato commented:
Update:
* added `fixICS.pl` to fix `Europe/Dublin` timezone in `timezones.ics`
* added usage of the script to Readme
* updated `timezones.ics` using the script. Now there is no difference in `TZID:Europe/Dublin` section. (See `timezones.ics`, new lines 3727-3746)
